### PR TITLE
Add composable index templates

### PR DESCRIPTION
**Pull request by ashwinkumar12345**
### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by aetter — APPROVED** [review]
Looks good to me. @keithhc2, any thoughts?

File: _opensearch/index-templates.md [hunk 1]
```diff
@@ -184,19 +175,233 @@ PUT _index_template/template-02
...
+    "component_template_2"
+  ],
+  "version": 3,
+  "_meta": {
+    "description": "using component templates"
+  }
+}
+```
+
+If you create an index named `logs-2020-01-01`, you can see that it derives it’s mappings and settings from both the component templates:
```
Comment (aetter) [inline _opensearch/index-templates.md:283]: its mappings

File: _opensearch/index-templates.md [hunk 2]
```diff
@@ -184,19 +175,233 @@ PUT _index_template/template-02
...
+  "template": {
+    "settings" : {
+        "index.number_of_shards" : 3
+    }
+  },
+  "composed_of": ["component_template_1", "component_template_2"]
+}
+```
+
+The `_simulate` API returns the final settings, mappings, and aliases that will be applied to indices that match the index pattern. You can also see any overlapping templates whose configuration is superseded by the simulated template body or higher-priority templates:
```
Comment (aetter) [inline _opensearch/index-templates.md:364]: I don't think we need the hyphen on higher priority.

File: _opensearch/index-templates.md [hunk 3]
```diff
@@ -184,19 +175,233 @@ PUT _index_template/template-02
...
+
+- If you have duplication between index templates, storing these index templates results in a bigger cluster state.
+- If you want to make a change across all your index templates, you have to manually make the change for each template.
+- If an index matches multiple templates, OpenSearch might merge the templates in an unexpected way that you discover only after an index is created.
+
+You can use composable index templates to overcome these challenges. Composable index templates let you abstract common settings, mappings, and aliases into a reusable building block called a component template.
+
+You can combine component templates to compose an index template.
+
+Settings and mappings that you specify directly in the [create index]({{site.url}}{{site.baseurl}}/opensearch/rest-api/create-index/) request overrides any settings or mappings specified in an index template and its component templates.
```
Comment (keithhc2) [inline _opensearch/index-templates.md:198]: I think this should be "specify directly in the **override** any settings..."

File: _opensearch/index-templates.md [hunk 4]
```diff
@@ -184,19 +175,233 @@ PUT _index_template/template-02
...
+    }
+  }
+}
+```
+
+### Use component templates to create an index template
+
+When creating index templates, you need to include the component templates in a `composed_of` list.
+
+OpenSearch applies the component templates in the order in which you specify them within the index template. The setting, mappings, and aliases that you specify inside the index template are applied last.
```
Comment (keithhc2) [inline _opensearch/index-templates.md:243]: I think we want "setting" to be plural here?

**Review by keithhc2 — COMMENTED** [review]
Everything looks good to me. Just found a couple typos.

File: _opensearch/index-templates.md (hunk 1 above)
Comment (ashwinkumar12345) [reply _opensearch/index-templates.md:283]: Fixed.

File: _opensearch/index-templates.md (hunk 2 above)
Comment (ashwinkumar12345) [reply _opensearch/index-templates.md:364]: Fixed.

File: _opensearch/index-templates.md (hunk 4 above)
Comment (ashwinkumar12345) [reply _opensearch/index-templates.md:243]: Fixed.

File: _opensearch/index-templates.md (hunk 3 above)
Comment (ashwinkumar12345) [reply _opensearch/index-templates.md:198]: Fixed.

**Pull request merged**